middleware/src/middleware.go

```diff
@@ -254,6 +254,31 @@ func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessag
 	}
 }
 
+// Hostname returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Hostname(args rpcmessages.HostnameArgs) (rpcmessages.GenericResponse, error) {
+	switch args.Method {
+	case rpcmessages.GetHostname:
+		log.Println("Getting the hostname via the config script")
+		out, err := middleware.runBBBConfigScript("get", "hostname")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.SetHostname:
+		log.Println("Setting the hostname via the config script")
+		out, err := middleware.runBBBConfigScript("set", "hostname"+" "+args.Name)
```

Comment:
Small nit:
`"hostname"+" "+args.Name` was visually hard for me to parse where all the pluses and quotes belong
Could you do it like in other places please? `"hostname" + " " + args.Name`
I think having spaces in between makes it a lot more readable

Reply:
Wanted this as well, but the _automatic `gofmt` on saving_ of my IDE changes this back to  `"hostname"+" "+args.Name`... Editing it manually in e.g. `vim` would only help until I edit it next time in the IDE... I'd opt to go with `gofmt` here, even if I don't agree.